### PR TITLE
Add "useSingleRequestParameter" option to Rust client generator

### PR DESCRIPTION
**Pull request by wing328**
- Add "useSingleRequestParameter" option to Rust client generator so that the method parameters will be grouped in a single struct
- Tested locally and the result is good.

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.3.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

cc @frol (2017/07) @farcaller (2017/08) @bjgill (2017/12) @richardwhiuk (2019/07) @paladinzh (2020/05)

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache [hunk 1]
```diff
@@ -31,7 +31,23 @@ pub trait {{{classname}}} {
 impl {{{classname}}} for {{{classname}}}Client {
 {{#operations}}
 {{#operation}}
+    {{#vendorExtensions.x-group-parameters}}
+    #[derive(Default)]
+    struct {{{operationIdCamelCase}}}Params {
```
Comment (pbeyeler) [inline modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache]: Should be public

File: modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache [hunk 2]
```diff
@@ -31,7 +31,23 @@ pub trait {{{classname}}} {
...
+        {{#allParams}}
+        {{{paramName}}}: {{^required}}Option<{{/required}}{{#required}}{{#isNullable}}Option<{{/isNullable}}{{/required}}{{#isString}}&str{{/isString}}{{#isUuid}}&str{{/isUuid}}{{^isString}}{{^isUuid}}{{^isPrimitiveType}}{{^isContainer}}crate::models::{{/isContainer}}{{/isPrimitiveType}}{{{dataType}}}{{/isUuid}}{{/isString}}{{^required}}>{{/required}}{{#required}}{{#isNullable}}>{{/isNullable}}{{/required}}{{#hasMore}}, {{/hasMore}}
```
Comment (pbeyeler) [inline modules/openapi-generator/src/main/resources/rust/reqwest/api.mustache]: Should be public

**Comment by wing328** [comment]
OK. Will update those.

**Comment by wing328** [comment]
Submitted https://github.com/OpenAPITools/openapi-generator/pull/6381

cc @pbeyeler

**Comment by pbeyeler** [comment]
@wing328 Thank you!

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/pull/6381 merged into master. Thanks for reporting the issue and suggesting the fix.

**Comment by wing328** [comment]
@pbeyeler I've filed https://github.com/OpenAPITools/openapi-generator/pull/6453 to export the struct. Please take a look when you've time.

**Comment by pbeyeler** [comment]
@wing328 Looks good to me.

However, I used the Stripe API spec (https://github.com/stripe/openapi/raw/master/openapi/spec3.yaml) for testing, which is quite complex. Trying to compile the resulting code resulted in 1256 errors (Rust 1.43.0). This also happens with v4.3.1. Most of these errors boil down to the same 4 or 5 causes. If you have time, I'd really appreciate if you could fix these errors as well. Thank you!

**Pull request merged**